edgepi-pwm.js

```diff
@@ -0,0 +1,110 @@
+module.exports = function (RED) {
+  const rpc = require("@edgepi-cloud/edgepi-rpc");
+
+  function PWMNode(config) {
+    RED.nodes.createNode(this, config);
+    const node = this;
+    const pwmPin = rpc.PWMPins[config.pwmPin];
+
+    setInitialConfigs(config, node)
+      .then((pwm) => {
+        node.on("input", async function (msg, send, done) {
+          node.status({ fill: "green", shape: "dot", text: "input received" });
+          try {
+            await updateConfigsOnInput(msg, pwm, config);
+            msg = {
+              payload: await pwm.getDutyCycle(pwmPin),
+              pwmPin: pwmPin,
+              frequency: await pwm.getFrequency(pwmPin),
+              polarity: await pwm.getPolarity(pwmPin),
+              enabled: await pwm.getEnabled(pwmPin),
+            };
+          } catch (error) {
+            console.error(error);
+            msg.payload = error;
+          }
+          send(msg);
+          if (done) done();
+        });
+
+        node.on("close", async () => {
+          try {
+            await pwm.close(pwmPin);
+          } catch (error) {
+            console.error(error);
+          }
+        });
+      })
+      .catch((error) => {
+        console.error(error);
+        node.status({
+          fill: "red",
+          shape: "ring",
+          text: "Initialization error",
+        });
+      });
+
+    async function setInitialConfigs(config, node) {
+      const transport =
+        config.transport === "Network"
+          ? `tcp://${config.tcpAddress}:${config.tcpPort}`
+          : "ipc:///tmp/edgepi.pipe";
+      const configArg = {
+        pwmNum: config.pwmPin,
+        frequency: parseFloat(config.frequency),
+        dutyCycle: parseFloat(config.dutyCycle) / 100,
+        polarity: config.polarity,
+      };
+
+      try {
+        const pwm = new rpc.PWMService(transport);
+        await pwm.initPwm(pwmPin);
+        await pwm.setConfig(configArg);
+        if (config.state === "enabled") {
+          await pwm.enable(pwmPin);
+        } else {
+          await pwm.disable(pwmPin);
+        }
+        return pwm;
+      } catch (error) {
+        error.message =
+          "Error setting initial configurations: " + error.message;
+        throw error;
+      }
+    }
+
+    async function updateConfigsOnInput(msg, pwm, config) {
+      let inputConfig = { pwmNum: pwmPin };
+      const { frequency, payload, polarity, enabled } = msg;
+
+      if (frequency) {
+        inputConfig.frequency = parseFloat(frequency);
+      }
+      if (payload) {
+        inputConfig.dutyCycle = parseFloat(payload) / 100;
+      }
+      if (polarity) {
+        inputConfig.polarity = rpc.Polarity[polarity];
+      }
+      if (enabled !== undefined) {
+        enabled ? await pwm.enable(pwmPin) : await pwm.disable(pwmPin);
+      }
```

Comment:
does it make sense to set pwm to disabled and then run setconfig on it?

Reply:
For the first time the configurations are set, setconfig needs to be called before enable/ disable. Since configurations might be set for the first time from input, I think it makes the most sense to have setconfig run first.

Reply:
I think based on enabled value you should have different branches. 
if enabled is true, set config and then enable
if enabled is false, then just disable. no need to call set config.